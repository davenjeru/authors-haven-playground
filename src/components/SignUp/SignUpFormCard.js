/* eslint-disable jsx-a11y/anchor-is-valid */
import { Link } from 'react-router-dom';
import React from 'react';
import FormInputField from '../commons/FormInputField';
import Loader from '../commons/Loader';

const SignUpFormCard = ({
  onFieldChange,
  onSubmit,
  onPasswordInputFocus,
  onPasswordInputBlur, onSignUpButtonBlur, preventSubmitOnEnter,
  getPasswordFieldClassName, shouldDisplayErrorMessage, passwordErrorMessage, isSubmitting,
}) => (
  <div
    className="card text-center mx-auto sign-up-form-card"
  >
    <div className="card-header">
      <h2>Sign Up</h2>
    </div>
    {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
    <form
      onSubmit={onSubmit}
      onKeyPress={preventSubmitOnEnter}
    >
      <div className="row">
        <p className="col-12">You could sign up with...</p>
        <div className="col-12">
          <button type="button" className="btn btn-primary">
            <a>
              <i className="fa fa-facebook fa-2x" aria-hidden="true" />
            </a>
          </button>
          <button type="button" className="btn btn-danger ml-3">
            <a>
              <i className="fa fa-google fa-2x" aria-hidden="true" />
            </a>
          </button>
        </div>

        <hr
          style={{ width: '100%' }}
        />
        <p className="col-12">Or</p>
      </div>
      <div className="px-5">
        <div className="form-group">
          <FormInputField
            type="email"
            required
            placeholder="Email Address"
            name="email"
            onChange={onFieldChange}
          />
        </div>
        <div className="form-group">
          <FormInputField
            type="text"
            required
            className="form-control"
            placeholder="Username"
            name="username"
            onChange={onFieldChange}
          />
        </div>
        <div className="form-group">
          <FormInputField
            type="password"
            required
            placeholder="Password"
            name="password"
            onChange={onFieldChange}
            onFocus={onPasswordInputFocus}
            onBlur={onPasswordInputBlur}
            className={getPasswordFieldClassName()}
          />
        </div>
        <div className="form-group">
          <FormInputField
            type="password"
            required
            placeholder="Confirm Password"
            name="confirmPassword"
            onChange={onFieldChange}
            onFocus={onPasswordInputFocus}
            onBlur={onPasswordInputBlur}
            className={getPasswordFieldClassName()}
          />
        </div>
      </div>
      {shouldDisplayErrorMessage
        ? <p className="mb-2 mx-5 p-1 alert alert-danger">{passwordErrorMessage}</p>
        : ''}
      {isSubmitting ? <Loader className="sign-up-loader" /> : ''}
      <div className="text-center">
        <button
          type="submit"
          onBlur={onSignUpButtonBlur}
          disabled={isSubmitting}
          className="btn btn-primary"
        >
          {isSubmitting ? 'Signing up...' : 'Sign Up'}
        </button>
        <p>
          Already have an account? Login
          <Link to="/login"> here</Link>
        </p>
      </div>
    </form>
  </div>
);


export default SignUpFormCard;
