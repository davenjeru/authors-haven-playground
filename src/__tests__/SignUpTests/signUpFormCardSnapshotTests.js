import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import SignUpFormCard from '../../components/SignUp/SignUpFormCard';

const propFunctionsMock = jest.fn();

const signUpFormCardPropsInitial = {
  onFieldChange: propFunctionsMock,
  onPasswordInputBlur: propFunctionsMock,
  getPasswordFieldClassName: propFunctionsMock,
  onPasswordInputFocus: propFunctionsMock,
  onSubmit: propFunctionsMock,
  onSignUpButtonBlur: propFunctionsMock,
  preventSubmitOnEnter: propFunctionsMock,
  isSubmitting: false,
  shouldDisplayErrorMessage: false,
  passwordErrorMessage: 'Passwords do not match',
};

describe('The SignUpFormCard', () => {
  it('should not regress - what is first seen', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <SignUpFormCard {...signUpFormCardPropsInitial} />
      </MemoryRouter>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should not regress - what is seen on submit with passwords that do not match', () => {
    const signUpFormCardPropsOnSubmitError = {
      ...signUpFormCardPropsInitial,
      shouldDisplayErrorMessage: true,
    };
    const tree = renderer.create(
      <MemoryRouter>
        <SignUpFormCard {...signUpFormCardPropsOnSubmitError} />
      </MemoryRouter>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should not regress - what is seen when submitting', () => {
    const signUpFormCardPropsWhileSubmitting = {
      ...signUpFormCardPropsInitial,
      isSubmitting: true,
    };
    const tree = renderer.create(
      <MemoryRouter>
        <SignUpFormCard {...signUpFormCardPropsWhileSubmitting} />
      </MemoryRouter>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
