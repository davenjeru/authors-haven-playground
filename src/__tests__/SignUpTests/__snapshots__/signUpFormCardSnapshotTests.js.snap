// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignUpFormCard should not regress - what is first seen 1`] = `
<div
  className="card text-center mx-auto sign-up-form-card"
>
  <div
    className="card-header"
  >
    <h2>
      Sign Up
    </h2>
  </div>
  <form
    onKeyPress={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <p
        className="col-12"
      >
        You could sign up with...
      </p>
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary"
          type="button"
        >
          <a>
            <i
              aria-hidden="true"
              className="fa fa-facebook fa-2x"
            />
          </a>
        </button>
        <button
          className="btn btn-danger ml-3"
          type="button"
        >
          <a>
            <i
              aria-hidden="true"
              className="fa fa-google fa-2x"
            />
          </a>
        </button>
      </div>
      <hr
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <p
        className="col-12"
      >
        Or
      </p>
    </div>
    <div
      className="px-5"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    
    
    <div
      className="text-center"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onBlur={[Function]}
        type="submit"
      >
        Sign Up
      </button>
      <p>
        Already have an account? Login
        <a
          href="/login"
          onClick={[Function]}
        >
           here
        </a>
      </p>
    </div>
  </form>
</div>
`;

exports[`The SignUpFormCard should not regress - what is seen on submit with passwords that do not match 1`] = `
<div
  className="card text-center mx-auto sign-up-form-card"
>
  <div
    className="card-header"
  >
    <h2>
      Sign Up
    </h2>
  </div>
  <form
    onKeyPress={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <p
        className="col-12"
      >
        You could sign up with...
      </p>
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary"
          type="button"
        >
          <a>
            <i
              aria-hidden="true"
              className="fa fa-facebook fa-2x"
            />
          </a>
        </button>
        <button
          className="btn btn-danger ml-3"
          type="button"
        >
          <a>
            <i
              aria-hidden="true"
              className="fa fa-google fa-2x"
            />
          </a>
        </button>
      </div>
      <hr
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <p
        className="col-12"
      >
        Or
      </p>
    </div>
    <div
      className="px-5"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    <p
      className="mb-2 mx-5 p-1 alert alert-danger"
    >
      Passwords do not match
    </p>
    
    <div
      className="text-center"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onBlur={[Function]}
        type="submit"
      >
        Sign Up
      </button>
      <p>
        Already have an account? Login
        <a
          href="/login"
          onClick={[Function]}
        >
           here
        </a>
      </p>
    </div>
  </form>
</div>
`;

exports[`The SignUpFormCard should not regress - what is seen when submitting 1`] = `
<div
  className="card text-center mx-auto sign-up-form-card"
>
  <div
    className="card-header"
  >
    <h2>
      Sign Up
    </h2>
  </div>
  <form
    onKeyPress={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <p
        className="col-12"
      >
        You could sign up with...
      </p>
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary"
          type="button"
        >
          <a>
            <i
              aria-hidden="true"
              className="fa fa-facebook fa-2x"
            />
          </a>
        </button>
        <button
          className="btn btn-danger ml-3"
          type="button"
        >
          <a>
            <i
              aria-hidden="true"
              className="fa fa-google fa-2x"
            />
          </a>
        </button>
      </div>
      <hr
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <p
        className="col-12"
      >
        Or
      </p>
    </div>
    <div
      className="px-5"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control undefined"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    
    <img
      alt="Loading..."
      className="sign-up-loader"
      src={
        Object {
          "0": "l",
          "1": "o",
          "2": "a",
          "3": "d",
          "4": "e",
          "5": "r",
          "6": ".",
          "7": "s",
          "8": "v",
          "9": "g",
          "default": "loader.svg",
        }
      }
    />
    <div
      className="text-center"
    >
      <button
        className="btn btn-primary"
        disabled={true}
        onBlur={[Function]}
        type="submit"
      >
        Signing up...
      </button>
      <p>
        Already have an account? Login
        <a
          href="/login"
          onClick={[Function]}
        >
           here
        </a>
      </p>
    </div>
  </form>
</div>
`;
